#!/usr/bin/env python
PACKAGE = "teleop_client"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("send_latency", double_t, 0, "The latency added to sending cmd_vel's", 0.0, 0.0, 4.0)
gen.add("receive_latency", double_t, 0, "The latency added to receiving telemetry", 0.0, 0.0, 4.0)
gen.add("use_one_latency", bool_t, 0, "If enabled, only the send_latency will be used on both directions of communication.", True)

# size_enum = gen.enum([ gen.const("Small", int_t, 0, "A small constant"),
#                   gen.const("Medium", int_t, 1, "A medium constant"),
#                   gen.const("Large", int_t, 2, "A large constant"),
#                   gen.const("ExtraLarge", int_t, 3, "An extra large constant") ],
#                   "An enum to set size")

# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "latency_reduction", "LatencyReduction"))
